src/assets.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Seahorse library.
+
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! The assets module.
+//!
+//! This module defines [Asset], [AssetEditor], and [Assets], which provide CURD (create, read,
+//! update, and delete) operations, with the use of [KeyValueStore] to control the assets resource.
+
+use crate::{asset_library::Icon, key_value_store::*, KeystoreError, Ledger, MintInfo};
+use jf_cap::structs::{AssetCode, AssetDefinition, AssetPolicy};
+use serde::{Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
+
+/// An asset with its code as the primary key.
+#[derive(Clone, Deserialize, Serialize)]
```

Comment:
We should derive `Debug` here. Is there some type in this that doesn't implement `Debug`?